system/Helpers/array_helper.php

```diff
@@ -103,9 +103,9 @@ function _array_search_dot(array $indexes, array $array)
     /**
      * Returns the value of an element at a key in an array of uncertain depth.
      *
-     * @param mixed $key
+     * @param float|int|string $key
```

Comment:
The key types in PHP are only int or string. If an other typed value comes, it will be cast.
https://www.php.net/manual/en/language.types.array.php#language.types.array.syntax

Therefore it should be `int|string`?

Reply:
I got it and modified.